pkg/volume/rbd/rbd.go

```diff
@@ -486,16 +518,83 @@ func (c *rbdUnmounter) TearDownAt(dir string) error {
 	return diskTearDown(c.manager, *c, dir, c.mounter)
 }
 
-func getVolumeSource(
-	spec *volume.Spec) (*v1.RBDVolumeSource, bool, error) {
+func getVolumeSourceMonitors(spec *volume.Spec) ([]string, error) {
+	if spec.Volume != nil && spec.Volume.RBD != nil {
+		return spec.Volume.RBD.CephMonitors, nil
+	} else if spec.PersistentVolume != nil &&
+		spec.PersistentVolume.Spec.RBD != nil {
+		return spec.PersistentVolume.Spec.RBD.CephMonitors, nil
+	}
+
+	return nil, fmt.Errorf("Spec does not reference a RBD volume type")
```

Comment:
more specific about the error, e.g. "spec does not reference a RBD volume typee: missing monitors", and for those below too.

Reply:
we check here only for volume spec whether it is regular or persistent volume. We do not check for presence of Monitors, it is done by a provisioner. I can change the error to something like "Spec does not reference to RBD volume or RBD Persistent volume types".

Reply:
this is fine as is